shell/src/main/java/alluxio/cli/fsadmin/pathconf/ShowCommand.java

```diff
@@ -0,0 +1,143 @@
+
+import alluxio.AlluxioURI;
+import alluxio.cli.CommandUtils;
+import alluxio.cli.fsadmin.command.AbstractFsAdminCommand;
+import alluxio.cli.fsadmin.command.Context;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.AlluxioProperties;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.path.PathConfiguration;
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.wire.Configuration;
+import alluxio.wire.Property;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Show path level configurations.
+ *
+ * <p>
+ * It has two modes:
+ * 1. only show configurations for the specific path;
+ * 2. show all configurations applicable to the path.
+ * <p>
+ * For example, suppose there are two path level configurations:
+ * /a: property1=value1
+ * /a/b: property2=value2
+ * Then in mode 1, only property2=value2 will be shown for /a/b,
+ * but in mode 2, since /a is the prefix of /a/b, both properties will be shown for /a/b.
+ */
+public final class ShowCommand extends AbstractFsAdminCommand {
+  public static final String RESOLVE_OPTION_NAME = "resolve";
+
+  private static final Option RESOLVE_OPTION =
+      Option.builder()
+          .longOpt(RESOLVE_OPTION_NAME)
+          .required(false)
+          .hasArg(false)
+          .desc("Whether to show all path level configurations applicable to the path")
+          .build();
+
+  /**
+   * @param context     fsadmin command context
+   * @param alluxioConf Alluxio configuration
+   */
+  public ShowCommand(Context context, AlluxioConfiguration alluxioConf) {
+    super(context);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "show";
+  }
+
+  @Override
+  public Options getOptions() {
+    return new Options()
+        .addOption(RESOLVE_OPTION);
+  }
+
+  @Override
+  public void validateArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
+  }
+
+  private String format(String key, String value) {
+    return key + " = " + value;
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    String targetPath = cl.getArgs()[1];
+    Configuration configuration = mMetaConfigClient.getConfiguration();
+
+    if (cl.hasOption(RESOLVE_OPTION_NAME)) {
+      Set<PropertyKey> pathPropertyKeys = new HashSet<>();
+      Map<String, AlluxioConfiguration> pathConfMap = new HashMap<>();
+
+      configuration.getPathConf().forEach((path, conf) -> {
+        AlluxioProperties properties = new AlluxioProperties();
+        conf.forEach(property -> {
+          pathPropertyKeys.add(PropertyKey.fromString(property.getName()));
```

Comment:
consider saving `PropertyKey.fromString(property.getName())` to a local variable since we use it twice here